### PR TITLE
Fix possible freeze in SemanticHighlightReconcilerStrategy

### PR DESCRIPTION
**Pull request by vrubezhny**
Getting rid of infinite waiting in SemanticHighlightReconcilerStrategy

### CODE REVIEW

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/semanticTokens/SemanticHighlightReconcilerStrategy.java [hunk 1]
```diff
@@ -82,6 +84,8 @@
 public class SemanticHighlightReconcilerStrategy
 		implements IReconcilingStrategy, IReconcilingStrategyExtension, ITextPresentationListener {
 
+    private static final int FULL_RECONCIL_TIMEOUT_S = 10;
```
Comment (rubenporras) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/semanticTokens/SemanticHighlightReconcilerStrategy.java:87]: should this not be `FULL_RECONCILE_TIMEOUT_S`?

**Comment by rubenporras** [comment]
I guess it is. I expect that in practice by the time 10 seconds have passed the request should have been answered or the user has typed again (and the the requests has been cancelled).

**Comment by rubenporras** [comment]
> @rubenporras Yet another question: should we interrupt current thread in case of `InterruptedException` - https://github.com/eclipse/lsp4e/blob/master/org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/semanticTokens/SemanticHighlightReconcilerStrategy.java#L298 by changing the catch block to:
> 
> ```
> 			} catch (ExecutionException e) {
> 				LanguageServerPlugin.logError(e);
> 			} catch (InterruptedException) {
> 				LanguageServerPlugin.logError(e);
> 				Thread.currentThread().interrupt();
> 			}
> ```
> 
> * as it's usually done in hovers, content assist, etc. - or we need this thread to stay uninterrupted?

It was an oversight not to add `Thread.currentThread().interrupt()`, so let us add it (even thought it is not clear what is the advantage of doing so, I think we should follow the pattern we already have)

**Comment by vrubezhny** [comment]
> It was an oversight not to add `Thread.currentThread().interrupt()`, so let us add it (even thought it is not clear what is the advantage of doing so, I think we should follow the pattern we already have)

OK.

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/semanticTokens/SemanticHighlightReconcilerStrategy.java (hunk 1 above)
Comment (vrubezhny) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/semanticTokens/SemanticHighlightReconcilerStrategy.java]: ACK

**Review by rubenporras — APPROVED** [review]

**Comment by mickaelistria** [comment]
Thanks!

**Pull request merged**